florist/api/routes/server/job.py

```diff
@@ -45,3 +45,29 @@ async def new_job(request: Request, job: Job = Body(...)) -> Dict[str, Any]:  #
     assert isinstance(created_job, dict)
 
     return created_job
+
+
+@router.get(path="/{status}", response_description="List jobs with the specified status", response_model=List[Job])
+async def list_jobs_with_status(status: JobStatus, request: Request) -> List[Dict[str, Any]]:
+    """
+    List jobs with specified status.
+
+    Fetches list of Job with max length MAX_RECORDS_TO_FETCH.
+
+    :param status: (JobStatus) The status of jobs to query the Job DB for.
+    :param request: (fastapi.Request) the FastAPI request object.
+    :return: (List[Dict[str, Any]]) A list where each entry is a dictionary with the attributes
+        of a Job instance with the specified status.
+    :raises: (HTTPException) status 400 if status is not a member of the JobStatus Enum.
+    """
+    status = jsonable_encoder(status)
+
+    assert isinstance(status, str)
+    if status not in JobStatus._member_names_:
+        msg = f"status is not valid. status: {status}."
+        raise HTTPException(status_code=400, detail=msg)
```

Comment:
I'm thinking if it will error by default if you pass in an invalid job status because of the `status: JobStatus` declaration in the parameters. Have you tried to pass in an invalid job status without the lines above being present?

Reply:
I thought so as well but if I comment this out the corresponding test fails. Any ideas?

Reply:
FastAPI only lets you select from a drop down with valid keys (which may be a good indication they do handle it) of the enum so I can. I will give it a shot with postman and loop back

Reply:
You are right! I forgot I were just calling the path operation function so obviously the test wont pass because FastAPI is going to handle it at the endpoint level. So I removed the code you flagged as well the aforementioned test

Reply:
Nice! It's great that FastAPI handles that for us. Less code is always better :)